src/mpid/ch4/subconfigure.m4

```diff
@@ -390,6 +381,14 @@ else
     AC_MSG_NOTICE([Using a non-symmetric heap])
 fi
 
+AC_CONFIG_FILES([
+src/mpid/ch4/src/mpid_ch4_net_array.c
+src/mpid/ch4/include/netmodpre.h
+src/mpid/ch4/include/shmpre.h
+])
+])dnl end AM_COND_IF(BUILD_CH4,...)
```

Comment:
@LenaO the diff makes it a little hard to see, but what I'm doing is effectively moving the end of the `AM_COND_IF(BUILD_CH4)` here so the `AC_CHECK_HEADERS` and `gl_FUNC_RANDOM_R` checks are inside it. There are 2 `BUILD_CH4` conditionals in this file. One sets some variables, the other runs checks. This is similar to how the ch3 subconfigure does it.